ow_power_system/src/power_system_node.cpp

```diff
@@ -278,44 +273,47 @@ void PowerSystemNode::injectFault (const string& fault_name,
       return;
     }
 
-    // TODO: Unspecified how to handle end of fault profile, which is
-    // unlikely.  For now, reuse the last entry.
-    if (index + m_profile_increment >= sequence.size()) {
-      ROS_WARN_STREAM_ONCE
-        (fault_name << ": reached end of fault profile, reusing last entry.");
-      // Probably unneeded, but makes index explicit.
-      index = sequence.size() - 1;
+    // If the current fault being utilized is high_power_draw,
+    // simply update wattage based on the current value of the HPD slider.
+    if (fault_name == FAULT_NAME_HPD_ACTIVATE)
+    {
+      ros::param::getCached("/faults/" + FAULT_NAME_HPD, hpd_wattage);
+      wattage += hpd_wattage;
+    } else
+    {
+      // TODO: Unspecified how to handle end of fault profile, which is
+      // unlikely.  For now, reuse the last entry.
+      if (index + m_profile_increment >= sequence.size()) {
+        ROS_WARN_STREAM_ONCE
+          (fault_name << ": reached end of fault profile, reusing last entry.");
+        // Probably unneeded, but makes index explicit.
+        index = sequence.size() - 1;
+      }
+      else index += m_profile_increment;
+
+      auto data = sequence[index];
+      wattage += data[MessageId::Watts];
+      voltage += data[MessageId::Volts];
+      temperature += data[MessageId::Centigrade];
     }
-    else index += m_profile_increment;
-
-    auto data = sequence[index];
-    wattage += data[MessageId::Watts];
-    voltage += data[MessageId::Volts];
-    temperature += data[MessageId::Centigrade];
   }
 }
 
 void PowerSystemNode::injectFaults(double& power,
 				   double& voltage,
 				   double& temperature)
 {
-  injectFault(FAULT_NAME_LOW_SOC,
-	      m_low_state_of_charge_power_failure_activated,
-              m_low_state_of_charge_power_failure_sequence,
-	      m_low_state_of_charge_power_failure_sequence_index,
-	      power, voltage, temperature);
-
-  injectFault(FAULT_NAME_ICL,
-	      m_instantaneous_capacity_loss_power_failure_activated,
-              m_instantaneous_capacity_loss_power_failure_sequence,
-              m_instantaneous_capacity_loss_power_failure_sequence_index,
-	      power, voltage, temperature);
-
-  injectFault(FAULT_NAME_THERMAL,
-	      m_thermal_power_failure_activated,
-	      m_thermal_power_failure_sequence,
-              m_thermal_power_failure_sequence_index,
-	      power, voltage, temperature);
+  // NOTE: Potential concerns:
+  //       The failure_sequence, failure_sequence_index,
+  //       voltage, and temperature
+  //       are no longer used as of JIRA 993. Should they
+  //       be left in for future use in case someone wants
+  //       to create a fault profile with additional modifiers?
+  injectFault(FAULT_NAME_HPD_ACTIVATE,
+        m_high_power_draw_power_failure_activated,
+        m_high_power_draw_power_failure_sequence,
+        m_high_power_draw_power_failure_sequence_index,
+        power, voltage, temperature);
```

Comment:
Yes, the sequence references should be removed, but leave the rest of the framework as is for now.  If my proposal for a new custom power fault profile is agreed on, this could be added in easily under a new Jira.

Reply:
I commented out the now-irrelevant parameters/variables without fully removing them. I imagine following additional feedback these could be removed completely, but until then I figured it would be best practice to leave them in as comments (though let me know if that isn't the case).